include/seqan3/range/container/constexpr_vector.hpp

```diff
@@ -0,0 +1,918 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ * \brief A constexpr vector implementation with dynamic size at compile time.
+ */
+
+#pragma once
+
+#include <array>
+#include <type_traits>
+
+#if SEQAN3_WITH_CEREAL
+#include <cereal/types/array.hpp>
+#endif // SEQAN3_WITH_CEREAL
+
+#include <seqan3/core/concept/cereal.hpp>
+#include <seqan3/core/detail/int_types.hpp>
+#include <seqan3/core/metafunction/template_inspection.hpp>
+#include <seqan3/range/view/repeat_n.hpp>
+#include <seqan3/range/view/take.hpp>
+#include <seqan3/std/ranges>
+
+namespace seqan3
+{
+
+/*!\brief A constexpr vector implementation with dynamic size at compile time.
+ * \implements seqan3::reservable_container_concept
+ * \implements seqan3::Cerealisable
+ * \ingroup container
+ * \tparam value_type_ The underlying value type stored in the vector.
+ * \tparam N           The capacity of the constexpr vector.
+ */
+template <typename value_type_, size_t N>
+class constexpr_vector
+{
+public:
+
+    /*!\name Member types
+     * \{
+     */
+    using value_type      = value_type_;                      //!< The value_type type.
+    using reference       = value_type &;                     //!< The reference type.
+    using const_reference = const value_type &;               //!< The const_reference type.
+    using iterator        = value_type *;                     //!< The iterator type.
+    using const_iterator  = value_type const *;               //!< The const_iterator type.
+    using difference_type = detail::min_viable_uint_t<N + 1>; //!< The difference_type type.
```

Comment:
changed it to `std::make_signed_t<size_type>`. that fine?

Reply:
No. If you have a capacity of 200, your size_type will be uint8_t, but your difference_type will be int8_t which cannot hold all the possible differences. It would probably be correct to `make_signed_t<min_viable_uint_t<(capacity_ + 1) * 2>>`, but maybe you can also just use `ptrdiff_t` here since (in contrast to size) no variable of that type is being saved.